EIPS/rentcollector.md

```diff
@@ -0,0 +1,73 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Rent Collection for Blockchain Data
+    Author: Daniel A. Nagy <daniel@ethereum.org>, Zsolt Felfoldi <zsolt@ethereum.org>
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-11-01
+
+
+## Simple Summary
+Introducing a way to collect rent from accounts that is proportional 
+to the cost that they impose on the network by storing a large state
+over a long period of time.
+
+## Abstract
+A new global parameter called `rent` similar to `gasLimit` that can be 
+modified by miner vote is introduced. It is initialized at 0 and is 
+assumed to have been zero before its introduction. The miner validating 
+a particular block can increase rent in a bounded way at the cost of 
+forfeiting a proportional part of the mining reward. Similarly, if 
+`rent` is more than zero, one can decrease it in a bounded way and 
+receive the same additional mining reward that would need to be 
+forfeited to raise it back.
+
+Each contract account gets a new balance called `lease` used for paying 
+rent that is initialized at 0 and is assumed to have been zero before 
+its introduction. For externally controlled accounts, their `lease` 
+balance is the same as their normal balance. With each new block, an 
+account is chosen in a deterministically random fashion with a 
```

Comment:
Choosing accounts weighted by their storage usage is going to require another consensus disk structure - one that has to be updated for every new storage slot allocated or unallocated.

Reply:
Not necessarily. If miners are full nodes, they can keep track of contract sizes in a canonical way off-consensus; after all the information is part of consensus implicitly and it is cheap to keep it up to-date.

An alternative solution could assume that state tries are balanced (deliberately unbalanced tries would only increase expected rent due to Jensen's inequality) and the size of each contract is assumed to be an exponential function of the length of a deterministically random branch in their state trie. In this case, the taxed account would be picked with uniform probability, but the levied rent would be `rent` multiplied by the estimated size of the state, going down the state trie along a deterministically random branch.